src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -255,10 +255,12 @@ export default class ReimbursementRequestService {
 
     if (!oldReimbursementRequest) throw new NotFoundException('Reimbursement Request', requestId);
     if (oldReimbursementRequest.dateDeleted) throw new DeletedException('Reimbursement Request', requestId);
-    if (oldReimbursementRequest.recipientId !== submitter.userId)
-      throw new AccessDeniedException(
-        'You do not have access to delete this reimbursement request, only the creator can edit a reimbursement request'
-      );
+    try {
```

Comment:
I see your point of using try and catch here to give a specific error message in case, the submitter is not a finance team member. However, when we look at how we used `validateUserIsPartOfFinanceTeam` in other RR service functions, we let the error be thrown from the `validateUserIsPartOfFinanceTeam` function if the submitter of edit RR is not a member of the finance team.

This is a way to keep our service function code simple and easier to understand. It also makes sense to give a general error msg of `You are not a member of the finance team!` in the case, that the submitter of edit RR is not a member of the finance team. 

In other words, we just want to use `validateUserIsPartOfFinanceTeam` function here instead of adding try and catch. We can keep the check if the submitter is a creator after `validateUserIsPartOfFinanceTeam` function usage.